### DISCUSSION TITLE
react query status questions

### DISCUSSION THREAD

**Discussion by yj-ilevit-com** [TanStack/query#7183] [Q&A]
![제목 없음-2024-03-19-17444](https://github.com/TanStack/query/assets/145316388/d950010a-20df-484c-98b5-4bedeb15dd57)

Hi! I'm a developer who has been using react query for a while now. 

I have a question about react query status. 

First question, the picture is my drawing, is it correct that when the component is mounted, assuming staleTime: 7000, it has the above state?

0-3: fresh & active
3-6: fresh & inactive
6-7:  fresh & active
7~: stale & active

Second question: what should we call fresh, stale, active, inactive, paused, deleted (removed from garbage collection) collectively?? react query status?

**Comment by TkDodo** [comment]
1) yes
2) `status` is `pending | success | error`. There isn't really a word for what you are looking for, the things you've listed are also not mutually exclusive (as the images shows)

**Reply by yj-ilevit-com** [reply]
Thanks for the quick response!

I have additional questions. I would like to be more clear about the definition of the word.

1. active, inactive => query type (I took a guess based on [this](https://github.com/TanStack/query/blob/main/packages/query-core/src/utils.ts#L61))
2. fresh, stale, paused, inactive, fetching => query state ([Here](https://github.com/TanStack/query/blob/main/packages/query-devtools/src/utils.tsx#L18C6-L18C22) it's defined as query status and [here](https://github.com/TanStack/query/blob/main/packages/vue-query/src/devtools/utils.ts#L6-L12) it's defined as query state)

Can we think of it like this?

**Reply by TkDodo** [reply]
the devtools only group these things together to make them more visible. `paused` is part of the `fetchStatus` (idle/paused/fetching). query data can be stale or fresh; a query can be active or inactive; a query can be in a fetchStatus, and a query has a status. pretty much all of the combinations are possible.

**Reply by yj-ilevit-com** [reply]
Thank you for your kind explanation.

But it doesn't seem to have what I want to know, so I ask again.

I am wondering what should I call xxx and yyy.
> status: pending | success | error
> fetchStatus: idle | paused | fetching
> xxx: stale | fresh 
> yyy: active | inactive

**Reply by TkDodo** [reply]
sorry you can call it whatever you want 😂 . There is no "official wording". Not sure why that's so important to you 🤷‍♂️

**Reply by yj-ilevit-com** [reply]
I'm trying to explain react query to people at work, but I want to be more precise, so I asked for a formal representation.
I thought if xxx, yyy had an official wording, it would be easier for the listener to understand and easier for the explainer to explain.
Let's call xxx the data status and yyy the query type.

> status: pending | success | error
> fetchStatus: idle | paused | fetching
> xxx: stale | fresh
> yyy: active | inactive
